### PR TITLE
New Kendo UI Rating widget

### PR DESCRIPTION
**Pull request by jlchereau**
As requested by admin at http://kendoui-feedback.telerik.com/forums/127393-telerik-kendo-ui-feedback/suggestions/2895213-rating-control

### CODE REVIEW

**Comment by jlchereau** [comment]
IMPORTANT: Make sure styles/kendo.rating.css is generated or create according to LESS file, then run tests/rating/kendo.rating.html and tests/rating/kendo.rating.test.html.

File: src/kendo.rating.js [hunk 1]
```diff
@@ -0,0 +1,308 @@
+/* jshint browser: true, jquery: true */
+/* globals define: false */
+
+(function (f, define) {
+    'use strict';
+    define(['./vendor/kendo/kendo.binder'], f);
```
Comment (underlog) [inline src/kendo.rating.js]: You should change this definition to the one like this:

https://github.com/telerik/kendo-ui-core/blob/master/src/kendo.button.js#L2

File: tests/rating/chai-jquery.js [hunk 2]
```diff
@@ -0,0 +1,231 @@
+(function (chaiJquery) {
```
Comment (underlog) [inline tests/rating/chai-jquery.js]: I am afraid that we e can't accept the additional testing framework dependency in our suite; we already have qUnit in place.

File: tests/rating/jquery.simulate.js [hunk 3]
```diff
@@ -0,0 +1,331 @@
+ /*!
```
Comment (underlog) [inline tests/rating/jquery.simulate.js]: We have most of that available in tests/kendo-test-helpers.js

File: tests/rating/kendo.rating.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
+<!DOCTYPE html>
```
Comment (underlog) [inline tests/rating/kendo.rating.html]: Please check the format of our tests - they need to live in JavaScript files only

**Comment by underlog** [comment]
Hi,

thank you very much for your pull request. The widget implementation is correct - however, we can't accept the test suite as it does not fit our existing suite/runner and conventions.

**Comment by jlchereau** [comment]
Hi,

Assuming there is a genuine interest from Telerik to adopt this widget, I am more than happy to spend some time converting our mocha tests to qunit tests as requested.

Please give me some insights about the future of this contribution before I commit the time.

Also note that I have more controls I would be willing to contribute to kendo UI core, including a property grid, a media player, ... Our job/value is to build applications, not UI controls and I am looking forward to handing them over.

**Comment by underlog** [comment]
We are genuinely interested in accepting this pull request. What we need for that is:

- A QUint based test suite that matches our standard widget sets. You can use this as a starting point:
https://github.com/telerik/kendo-ui-core/tree/master/tests/textbox

- API reference document, like this one:
https://github.com/telerik/kendo-ui-core/blob/master/docs/api/javascript/ui/maskedtextbox.md

- Skin support. This should be fairly easy to implement. 

Once this is in place, we will make sure to include the widget in our distribution, and maintain it alongside the rest of the widgets. We will also create the necessary demos in the web site.  

Please notice however, that we will be able to integrate it after the Q3 release, due end of this month.

**Comment by underlog** [comment]
@jlchereau , feel free to reopen this thread in case you make any progress with the widget. Thanks!

**Pull request closed without merging**